### PR TITLE
support creation of integration responses for apigw method

### PR DESCRIPTION
**Pull request by pinzon**
As @calvernaz pointed out. The ApiGW Method wasn't creating correctly the Integration Responses. This PR addresses that.

Changes:

- [x] Refactor the test for the apigateway integration deployment
- [x] Resource model creates the integration response 
- [x] Change the template to Yaml for easiear reading

### CODE REVIEW

File: localstack/services/cloudformation/models/apigateway.py [hunk 1]
```diff
@@ -365,26 +372,30 @@ def _subresources(resource_id, resources, resource_type, func, stack_name):
...
-
-                if integration.get("Credentials"):
-                    kwargs["credentials"] = integration["Credentials"]
-
-                if integration.get("RequestParameters"):
-                    kwargs["requestParameters"] = integration["RequestParameters"]
+                integration_responses = []
+                if kwargs.get("integrationResponses"):
+                    integration_responses = kwargs["integrationResponses"]
+                    del kwargs["integrationResponses"]
```
Comment (simonrw) [inline localstack/services/cloudformation/models/apigateway.py:378]: nit: could this work? Then you don't have to don't have to check and delete from `kwargs`
```suggestion
                integration_responses = kwargs.pop("integrationResponses", [])
```

**Review by simonrw — CHANGES_REQUESTED** [review]
Looks good, but I'm keen to understand why the `integrationResponses` are not being validated

File: tests/integration/cloudformation/resources/test_apigateway.py [hunk 2]
```diff
@@ -191,17 +191,31 @@ def test_cfn_with_apigateway_resources(deploy_cfn_template, apigateway_client):
...
 
 
-# TODO: rework
+@pytest.mark.skip_snapshot_verify(
+    paths=[
+        "$..binaryMediaTypes",
+        "$..version",
+        "$..methodIntegration.cacheNamespace",
+        "$..methodIntegration.connectionType",
+        "$..methodIntegration.integrationResponses",
```
[the remark this answers is not in the document]
Comment (pinzon) [reply tests/integration/cloudformation/resources/test_apigateway.py:200]: The correct skip_path shoudl be `$..methodIntegration.integrationResponses.200.responseParameters ( {} )` but the snapshot lib crashes using that path. 

the problem is that our ApiGW get_method is not returning in the response the empty responseParameters dict. even though is specified [here](https://github.com/localstack/localstack/blob/e1b478ba738f8b1a7b7195b15d7313f54b7b6baf/localstack/services/cloudformation/models/apigateway.py#L386) 

But yeah this skip can be confusing. I'll add more parameters so we can avoid it.

Thanks for noticing @simonrw

**Review by simonrw — APPROVED** [review]
LGTM! Thanks for addressing my changes :+1:

**Pull request merged**